p4proto/kctrl/switchapi/README.md

```diff
@@ -0,0 +1,166 @@
+   * Triggered by SwitchSAI create_fdb_entry API. Creates FDB handle based on type of information received (Tx or Rx). Tx FDB learn can be from tunnel interface, physical interface or VLAN interface.
+   * Table entry is created with platform dependent (PD) API (switch_pd_l2_tx_forward_table_entry for Tx entry and switch_pd_l2_rx_forward_table_entry for Rx entry).
+   * Once the table entry succeeds, handle along with api_l2_info is cached in l2 hashtable with mac address as the key.
+ * **switch_api_l2_forward_delete**:
+   * Triggered by SwitchSAI delete_fdb_entry API. Deletes the table entry using platform dependent API based on type of handle(Tx/Rx) and deletes the cache from Judy array
+
+## RIF <a name="rif"></a>
+ * **switch_api_rif_create**: Handle is allocated and initialized for handle type SWITCH_HANDLE_TYPE_RIF and the port ID for which this function is triggered, is determined
+using platform independent API switch_pd_to_get_port_id and saved in rif_info structure.
+ * **switch_api_rif_delete**: Deletes the associated RIF handle
+
+## VRF <a name="vrf"></a>
+ * **switch_api_vrf_create**: Handle is allocated and initialized for handle type SWITCH_HANDLE_TYPE_VRF. Based on the VRF id attribute passed by SwitchSAI, check if VRF handle exists for this VRF id. If not, create the VRF handle and cache in Judy array with vrf id as key and return VRF handle to the SwitchSAI
+ * **switch_api_vrf_delete**: Retrieve the vrf_info for associated VRF handle, delete the cached information from Judy array and delete the handle
+
+## RMAC <a name="rmac"></a>
+ * **switch_api_router_mac_group_create**: Handle is allocated and initialized for handle type SWITCH_HANDLE_TYPE_RMAC.  Retrieves the rmac_info based on rmac handle and initalize the rmac list
+ * **switch_api_router_mac_add**: Iterates through all the MAC's in rmac_list and create an entry for each MAC
+ * **switch_api_router_mac_group_delete** - Iterates through all entries in rmac list amd delete the rmac entry and handle associated
+
+## Neighbor <a name="neighbor"></a>
+ * **switch_api_neighbor_create**: Handle is allocated and initialized for handle type SWITCH_HANDLE_TYPE_NEIGHBOR. Retrieves the nhop_handle from api_neighbor_info attribute passed by SwitchSAI. If nhop handle is valid, create the handle for neighbor and update the nexthop table and neighbor mod table using appropriate platform dependent (PD) API's
+ * **switch_api_neighbor_delete**: Retrieve the neighbor_info from neighbor handle and call the platform dependent (PD) API's to delete table entry from target backend and delete the handle
+
+## Nexthop <a name="nexthop"></a>
+ * **switch_api_nhop_create**: Handle is allocated and initialized for handle type SWITCH_HANDLE_TYPE_NHOP. Check if the nhop handle exists, if not, create one and get the nhop key based on nhop type. Once the handle is successfully created, cache the nhop info with key as nhop key in the hashtable and return nhop handle to SwitchSAI
+ * *Note*:
```

Comment:
this looks little unaligned, please check

Reply:
Alignment is fine here: 
![image](https://user-images.githubusercontent.com/22621220/181165214-bba7a0c8-b9b1-4cef-a85d-9e8a351a47dd.png)
Please let me know if you still see any issue